rpcserver.go

```diff
@@ -1557,6 +1558,53 @@ func handleExistsLiveTickets(s *rpcServer, cmd interface{},
 	return hex.EncodeToString([]byte(set)), nil
 }
 
+// handleExistsMempoolTxs implements the existsmempooltxs command.
+func handleExistsMempoolTxs(s *rpcServer, cmd interface{},
+	closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*dcrjson.ExistsMempoolTxsCmd)
+
+	txHashBlob, err := hex.DecodeString(c.TxHashBlob)
+	if err != nil {
+		return nil, &dcrjson.RPCError{
+			Code:    dcrjson.ErrRPCDecodeHexString,
+			Message: "bad transaction hash blob (unparseable)",
+		}
+	}
+
+	// It needs to be an exact number of hashes.
+	if len(txHashBlob)%32 != 0 {
+		return nil, &dcrjson.RPCError{
+			Code:    dcrjson.ErrRPCDecodeHexString,
+			Message: "bad transaction hash blob (bad length)",
+		}
+	}
+
+	hashesLen := len(txHashBlob) / 32
+	hashes := make([]*chainhash.Hash, hashesLen)
+	for i := 0; i < hashesLen; i++ {
+		hashes[i], err = chainhash.NewHash(
+			txHashBlob[i*chainhash.HashSize : (i+1)*chainhash.HashSize])
+	}
+
+	exists := s.server.txMemPool.HaveTransactions(hashes)
+	if len(exists) != hashesLen {
+		return nil, &dcrjson.RPCError{
+			Code:    dcrjson.ErrRPCDatabase,
+			Message: "output of ExistsMempoolTxs wrong size",
```

Comment:
Message: fmt.Sprintf("output of ExistsMempoolTxs wrong size: %d != %d", len(exists), hashesLen),